src/fqe/wavefunction.py

```diff
@@ -548,20 +549,21 @@ def apply_generated_unitary(self,
         else:
             base = self
 
-        max_expansion = max(30, expansion)
```

Comment:
If you remove the max expansion order, maybe add a check (raise a ValueError or something) if it is unreasonably large?

Reply:
I'll add a check for expansion being an integer.  That way the user can't input a np.nan, np.inf, float...something that would make the loop fail.